tests/test_physics/test_ansatz.py

```diff
@@ -355,3 +356,223 @@ def test_bargmann_Abc_to_phasespace_cov_means(self):
         assert math.allclose(new_state_means22[0], state_means)
         assert math.allclose(new_state_coeff1[0], 1.0)
         assert math.allclose(new_state_coeff22[0], 1.0)
+
+
+class TestDiffOpPolyExpAnsatz:
+    r"""
+    Tests the ``DiffOpPolyExpAnsatz`` class.
+    """
+
+    Abc_n1 = Abc_triple(1)
+    Abc_n2 = Abc_triple(2)
+    Abc_n3 = Abc_triple(3)
+
+    @pytest.mark.parametrize("triple", [Abc_n1, Abc_n2, Abc_n3])
+    def test_init(self, triple):
+        A, b, c = triple
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        assert np.allclose(ansatz.mat[0], A)
+        assert np.allclose(ansatz.vec[0], b)
+        assert np.allclose(ansatz.array[0], c)
+
+    def test_add(self):
+        A1, b1, c1 = Abc_triple(5)
+        A2, b2, c2 = Abc_triple(5)
+
+        ansatz = DiffOpPolyExpAnsatz(A1, b1, c1)
+        ansatz2 = DiffOpPolyExpAnsatz(A2, b2, c2)
+        ansatz3 = ansatz + ansatz2
+
+        assert np.allclose(ansatz3.mat[0], A1)
+        assert np.allclose(ansatz3.vec[0], b1)
+        assert np.allclose(ansatz3.array[0], c1)
+        assert np.allclose(ansatz3.mat[1], A2)
+        assert np.allclose(ansatz3.vec[1], b2)
+        assert np.allclose(ansatz3.array[1], c2)
+
+    def test_mul(self):
+        A1, b1, c1 = Abc_triple(5)
+        A2, b2, c2 = Abc_triple(5)
+
+        ansatz = DiffOpPolyExpAnsatz(A1, b1, c1)
+        ansatz2 = DiffOpPolyExpAnsatz(A2, b2, c2)
+        ansatz3 = ansatz * ansatz2
+
+        assert np.allclose(ansatz3.mat[0], A1 + A2)
+        assert np.allclose(ansatz3.vec[0], b1 + b2)
+        assert np.allclose(ansatz3.array[0], c1 * c2)
+
+    def test_mul_scalar(self):
+        A, b, c = Abc_triple(5)
+        d = 0.1
+
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        ansatz2 = ansatz * d
+
+        assert np.allclose(ansatz2.mat[0], A)
+        assert np.allclose(ansatz2.vec[0], b)
+        assert np.allclose(ansatz2.array[0], d * c)
+
+    def test_truediv(self):
+        A1, b1, c1 = Abc_triple(5)
+        A2, b2, c2 = Abc_triple(5)
+
+        ansatz = DiffOpPolyExpAnsatz(A1, b1, c1)
+        ansatz2 = DiffOpPolyExpAnsatz(A2, b2, c2)
+        ansatz3 = ansatz / ansatz2
+
+        assert np.allclose(ansatz3.mat[0], A1 - A2)
+        assert np.allclose(ansatz3.vec[0], b1 - b2)
+        assert np.allclose(ansatz3.array[0], c1 / c2)
+
+    def test_truediv_scalar(self):
+        A, b, c = Abc_triple(5)
+        d = 0.1
+
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        ansatz2 = ansatz / d
+
+        assert np.allclose(ansatz2.mat[0], A)
+        assert np.allclose(ansatz2.vec[0], b)
+        assert np.allclose(ansatz2.array[0], c / d)
+
+    def test_call(self):
```

Comment:
What do you mean with analytically calculated one? When c is a vector the only way to calculate the value is by going through the call functions which also includes a call to `hermite_renorm`. I can create all that logic in the test, but it will look very similar to the `call` function.

Reply:
E.g. for the A,b,c corresponding to a two-mode squeezed state, we know the poly*exp ansatz for a c with all zeros and one 1 in it should should just be a Fock state, where the analytic expression is known.